GitUI/UserControls/WarningHeader.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using GitUI.CommandsDialogs;
+using Microsoft.VisualStudio.Threading;
+
+namespace GitUI.UserControls
+{
+    public partial class WarningHeader : UserControl
+    {
+        private enum State
+        {
+            None,
+            Bisect,
+            Rebase,
+            Merge,
+            Patch
+        }
+
+        private State _state = State.None;
+        private bool _hasConflicts = false;
+
+        public event EventHandler BisectEvent;
+
+        public WarningHeader()
+        {
+            InitializeComponent();
+        }
+
+        public async Task RefreshAsync(GitCommands.GitModule module)
+        {
+            // get the current state of the repo
+            await TaskScheduler.Default;
+
+            if (module.IsValidGitWorkingDir())
+            {
+                if (module.InTheMiddleOfBisect())
+                {
+                    await SetStateAsync(State.Bisect, false);
+                }
+                else
+                {
+                    bool conflicted = module.InTheMiddleOfConflictedMerge();
+
+                    if (module.InTheMiddleOfRebase())
+                    {
+                        await SetStateAsync(State.Rebase, conflicted);
+                    }
+                    else if (module.InTheMiddleOfMerge())
+                    {
+                        await SetStateAsync(State.Merge, conflicted);
+                    }
+                    else if (module.InTheMiddleOfPatch())
+                    {
+                        await SetStateAsync(State.Patch, conflicted);
+                    }
+                    else
+                    {
+                        await SetStateAsync(State.None, false);
+                    }
+                }
+            }
+
+            async Task SetStateAsync(State state, bool conflicted)
+            {
+                if ((state == _state) && (conflicted == _hasConflicts))
+                {
+                    // nothing to do
+                    return;
+                }
+
+                _state = state;
+                _hasConflicts = conflicted;
+
+                // now update in main UI thread
+                await this.SwitchToMainThreadAsync();
+
+                // remove old controls
+                ButtonContainer.Controls.Clear();
+
+                string action = "";
+
+                switch (_state)
+                {
+                    case State.Bisect:
+                        action = "Bisect";
```

Comment:
This string (and the others below) will need to be translated using `TranslationString`.

Reply:
Can I build the TranslationString as I currently am doing, or do they need to be defined in full?

Reply:
I think that a full sentence is better from a translation point of view.
You could do 2 sentences with and without conflicts with the 'string.Format()' format to replace the 'action': "{0} currently in progress with merge conflicts"

Reply:
Yes, that sounds perfect, thanks!